src/prted/pmix/pmix_server_queries.c

```diff
@@ -553,7 +553,12 @@ static void _query(int sd, short args, void *cbdata)
             } else if (0 == strcmp(q->keys[n], PMIX_JOB_SIZE)) {
                 jdata = prte_get_job_data_object(jobid);
                 if (NULL == jdata) {
-                    rc = PRTE_ERR_NOT_FOUND;
+                    ret = PRTE_ERR_NOT_FOUND;
+                    goto done;
+                }
+                /* Check if there are any entries in local proctable */
+                if (0 == jdata->num_local_procs) {
```

Comment:
I don't think this check is correct. We needed it for `PMIX_QUERY_LOCAL_PROC_TABLE` since it's focused on the local processes. For `PMIX_JOB_SIZE` we could be quering the job size of a non-local namespace, so we want to return what we know of it.